### DISCUSSION TITLE
📋 About Framework Design

### DISCUSSION THREAD

**Discussion by slhmy** [oj-lab/frontend#1] [Ideas]
Thanks @KanadeSiina for joining in the great work!
Briefly, we will copy the things in [oj-lab-example-front](https://github.com/OJ-lab/oj-lab-example-front), but in Typescript (instead of Javascript).

I have prepared some of the basic folder for this repository, so let's talk about the folders first.
`src/components`, `src/pages`, `src/routes` are folders we should touch first, they are very basic in usage.
Then we can move onto `src/themes`, `src/i18n` for future frontend customization.
Meanwhile`src/service` can also be done, but it is more complex, so we can push slower.

I will give some info for the dependencies we decide to use.
- [React Router](https://v5.reactrouter.com/web/guides/quick-start)
- [Redux](https://redux.js.org/)
- [mui](https://mui.com/zh/)
- [i18next](https://www.i18next.com/)
- [Mock.js](https://mockjs.com/)

Finally I suggest other thing if not listed should first follow the instruction provided by [Create React App](https://create-react-app.dev/).
**For each step of the related work, we should first raise an issue for it and trace it by milestone [Basic framework completion](https://github.com/OJ-lab/oj-lab-front/milestone/1).**

**Comment by zy6p** [comment]
Congratulations🎉
